src/apps/material-grid-automatic/MaterialGridAutomatic.tsx

```diff
@@ -0,0 +1,120 @@
+): MaterialGridAutomaticProps["amountOfMaterialsToShow"] {
+  const validIncrements: MaterialGridAutomaticProps["amountOfMaterialsToShow"][] =
+    [4, 8, 12, 16, 20, 24, 28, 32];
+
+  // Use the exact amount if we have fetched enough materials
+  if (fetchedItemCount >= amountOfMaterialsToShow) {
+    return amountOfMaterialsToShow;
+  }
+
+  // Use the largest valid increment if we have fetched less than the selected amount
+  let i = validIncrements.length - 1;
+  while (i >= 0) {
+    if (validIncrements[i] <= fetchedItemCount) {
+      return validIncrements[i];
+    }
+    i -= 1;
+  }
+  // Fallback for minimum increment
+  return 4;
+}
+
+export type MaterialGridAutomaticProps = {
+  cql: string;
+  title: string;
+  amountOfMaterialsToShow: 4 | 8 | 12 | 16 | 20 | 24 | 28 | 32;
+};
+const MaterialGridAutomatic: React.FC<MaterialGridAutomaticProps> = ({
+  cql: cqlProp,
+  title,
+  amountOfMaterialsToShow
+}) => {
+  const [cql, setCql] = useState<string>(cqlProp);
+  const [displayedItemCount, setDisplayedItemCount] = useState(
+    amountOfMaterialsToShow
+  );
+
+  const [showAllMaterials, setShowAllMaterials] = useState(false);
+
+  const cleanBranches = useGetCleanBranches();
+  const { data, isLoading } = useComplexSearchWithPaginationQuery({
+    cql,
+    offset: 0,
+    limit: amountOfMaterialsToShow,
+    filters: {
+      branchId: cleanBranches
+    }
+  });
+
+  useEffect(() => {
+    setCql(cql);
+  }, [cql]);
+
+  useEffect(() => {
+    if (data && data.complexSearch && data.complexSearch.works) {
+      const fetchedItemCount = data.complexSearch.works.length;
+      const newItemCount = calculateDisplayedItemCount(
+        fetchedItemCount,
+        amountOfMaterialsToShow
+      );
+      setDisplayedItemCount(newItemCount);
+    }
+  }, [data, amountOfMaterialsToShow]);
+
+  if (isLoading || !data) {
+    return <MaterialGridAutomaticSkeleton />;
+  }
+
+  const {
+    complexSearch: { works: resultWorks }
+  } = data as {
+    complexSearch: {
+      works: Work[];
+      hitcount: ComplexSearchWithPaginationQuery["complexSearch"]["hitcount"];
+    };
+  };
+
+  const initialVisibleItemCount = 8;
+  const showMoreButton = displayedItemCount > initialVisibleItemCount;
+
+  return (
+    <div className="material-grid">
+      <h2 className="material-grid__title">{title}</h2>
+      <ul className="material-grid__items">
+        {resultWorks.slice(0, displayedItemCount).map((material) => (
+          <li key={material.workId} className="material-grid__item">
+            <RecommendedMaterial wid={material.workId} />
+          </li>
+        ))}
+      </ul>
+
+      {showMoreButton && !showAllMaterials && (
+        <button
+          className="material-grid__show-more btn-primary btn-outline btn-medium"
+          data-show-more
+          aria-expanded="false"
+          type="button"
+          onClick={() => setShowAllMaterials(true)}
+        >
+          Vis alle
```

Comment:
This text should be passed as a prop and displayed using `useText()` like we do for other UI strings.like `et al.`.